### PR TITLE
Add parenthesis around lambda inside oneliner infix app

### LINKED ISSUE #2196: Piped lambda with if-then-else can break code
Issue created from [fantomas-online](https://fsprojects.github.io/fantomas-tools/#/fantomas/preview?data=N4KABGBEDGD2AmBTSAuKAbRAXM8CGAngPIBmA6oogNYAqsAcgK4C2YAFPGgCKGkXUBKMAF4AOgCdRAOwgQAllJzwJ02WAA%2BAPjAlGMgB5gAtNrkkwh4WAAMYLAAtEMgOxhE6AM6ILKmbK1gPARMzL6QADTgUF5YWAoA5h6oYADaURCgahCQACRYBAAOyGiQClgR6bK5AG546IzFqZVq1pFZVQCSUkiKAMpyAF7Ibe1gACzNYAC6lQC%2BIxmTuflFyaWKFe01dQ3JaaNgAIwLWZAAsnj6ADIKiFdO8Q6bo4cATNaTM2rzlZmneYVGpAnPBSDcpIhevlMM81Nt6o19qNXic4QBRbpg26w07oEiQT5zE5-OEA1YlABGsFg6Bx2RytQRe0mEAAzKjOlIvOIsAAxBR1eiIADu6GxHIgWHEDUJ32JSzJQKpNLpUAZO0RLPGEqgvUQzDkcHQsCkAEEsBjQSRwcMtSQ6l5ZbIfmoSVVFWtlbSOfDdmgke0AKw6yC9Ap4aCIABCiBIsHEiAACnhxHhmNhEOJVZLpYgnRAXbI3fSVkrqd6FYy-U0DgA2ENhiPR2Pxu6wYWZ6B4LxdaqwLtxE3Zuy5-NgQuLLYeynl1W%2BzUHZwN8ORmNxhMAVQKRXEXZ7Uj7A7kQ519s8efaX2d8qnpc9s596qZ-q1AA5l02162AMLobseb8TQ8KVGGgLB42HM9HUvIlfgVO8ZxVR8qwXUYAE4P1XFsEzOfUKUzSCHQvLIrwLG9-gQqAvTnJ9qwDLJDlaLVQxXZt10QQDjSkQjzzHCcwGLNVKMgajkI1ZkDkOY5mMbSNTRILBM0A5hmDwYcQOItRSPHcjSWE0TK3El9JJRGTWK-BM9QNI0TztIi%2BN0919IfQznxrF52TMpt5MU8QrMNGlbIODSHLg29AXvJDXLorVDjGEMuh6LAiCkGhxAIMg5CeU97JguUwooiLEIrKcUIkl5gy8uTxFgPR4C4dw5ANXz1NHPLrwKvSiqgMoaLK4yXnrZiLn0DoSBoRwpDRc9enseMsEy%2BBsq1MYPnasjOqc7r1nKMS3PotRDiXYbLi6Eg5H0IgdzwcDxDRfQCgTDwPGPbidUDNaSNg114O23q9piyT3xO-QACVEDgcR4EW5aDlW0KfvC8keo2AHUPaQ4MJB8HIfgEJ8PEUgOkU5gkh1Q4EaLX7kcgZhGHQOIxQhXl41U2JMxoCK0fK9p3hDHH43gM56cZ24WfENmWpDaB7BTCNfIAfWFOQlvsAl1p0zaSz%2B1HovRrJXmkg5zkuU1xFTYhxBuYCYbVnV4Y1-jBOWHXdr1nmDdM42RrNi2iCtuRgPxzMiZJsnYspydCppumGbkJnEHFyWOa592Bt5zzjd93gA%2BA4W44TpObql5iZblsDMyVlWnkjgTqaBf60-c3n4pBgA1DUowUeAElt4dX0%2BrTvqppGG910qjObg3Ku9y5eT0MDXq77pe%2Bru2VsH2RtKd%2Bu1kbif9q1V4htn-QuFgLAAHFsHux7EGe16%2B-ezeIG3xztZpgyD8B5Fju9kX463CjMaaAVAoyplAdgDwKVehpg4jSFgb07K8Udu-IS20v7-H6lPNQrxgbGyFKKIB2AOxOE5kUBq50pBZVeqabouFmAE3DgcKCmkt7Dyjl1T%2BLlv761wVjY2ABpSgBQxoTScF0WB6YbQ8Wgl9fKiNo5jzdrwj2ahWRMVPtNA0Hh7BPw3rXZ204qI8KwZPA6shWRG1GKGBImAzbxBYE4BaiAKRnAQLaFhuV5EdUUVwssUVVHpyyKyL2NjTRiniFIeeUhF4mn6FEm6jAEx0ESs4m6r1ZFsNfhwuuo9IolTMYfA4rJM7hIZpmKQGTqhtikPEBhBNKEKBoUBLJhjd7FT6uYrUrJW7-zjjaK4aYKT4F-LAF6dTCEJzaagrW6DuGBKKT-dorIZ42K4IHPAFJMDaMDvYXoBBFCXBmT4jafitoLMKaSbBFi2QnxscIxAojujOK6OAvAsT15eJQaczW5yP4BKue6G5PS-42OAR8qgNoPBm1qt0IUwFEBCwAQnW%2BT0XqtJ1CFWZ-z5mAq6cU0YrJ8HgpTBZRAGyPDQHEE1AUil4AbmoSaBq0A-ypkHEg75cih4KJHkogpBLlkhIETY0GeBEk1SoCcnlvi%2BX%2BIFdzYJahVoNilIaLA7ikDSvYd8KIUwRilA8LyF6yRWEgFmEAA)

#### Code

```fsharp
let dayOfWeekToNum (d: DayOfWeek) =
    int d
    |> fun x -> if x = 0 then 7 else x
    |> DayNum

```
            

#### Result

```fsharp
let dayOfWeekToNum (d: DayOfWeek) =
    int d |> fun x -> if x = 0 then 7 else x |> DayNum

```
            
#### Problem description

Meaning of code was changed due to `|>` having higher precedence than `else`.

#### Proposed solution

```fsharp
let dayOfWeekToNum (d: DayOfWeek) =
    int d |> (fun x -> if x = 0 then 7 else x) |> DayNum

```

#### Extra information

- [x] The formatted result breaks by code.
- [ ] The formatted result gives compiler warnings.
- [x] I or my company would be willing to help fix this.

#### Options

Fantomas master branch at 2022-04-06T12:09:12Z - 5cbe6a3d82c6eee12ba054b6a99ce20ef875e201

Default Fantomas configuration

<sub>Did you know that you can ignore files when formatting from fantomas-tool or the FAKE targets by using a [.fantomasignore file](https://github.com/fsprojects/fantomas/blob/master/docs/Documentation.md#ignore-files-fantomasignore)?</sub>

### PR DESCRIPTION
**Pull request by jindraivanek**
Fixes #2196

### CODE REVIEW

File: src/Fantomas/CodePrinter.fs [hunk 1]
```diff
@@ -1703,9 +1703,16 @@ and genExpr astContext synExpr ctx =
                 genExpr astContext e
                 +> sepSpace
                 +> col sepSpace es (fun (s, oe, e) ->
+                    let isLambda =
```
Comment (nojaf) [inline src/Fantomas/CodePrinter.fs:1706]: I think we can re-use

https://github.com/fsprojects/fantomas/blob/04d176a519106751750e29e5754075eb475e827c/src/Fantomas/SourceParser.fs#L1798-L1802

here.

**Review by nojaf — CHANGES_REQUESTED** [review]

**Comment by nojaf** [comment]
Thanks! Looks good, one little nitpick.
Cheers!

File: src/Fantomas/CodePrinter.fs (hunk 1 above)
Comment (jindraivanek) [reply src/Fantomas/CodePrinter.fs]: Thanks for pointing that out.

**Review by nojaf — APPROVED** [review]
Thanks!

**Pull request merged**